numpy/core/function_base.py

```diff
@@ -1,11 +1,25 @@
 from __future__ import division, absolute_import, print_function
 
+import warnings
+import operator
+
 __all__ = ['logspace', 'linspace']
 
 from . import numeric as _nx
 from .numeric import result_type, NaN, shares_memory, MAY_SHARE_BOUNDS, TooHardError
 
 
+def index_deprecate(i):
```

Comment:
Yes actually I was trying to find a better place where I could put this `index_deprecate` function like some sort of utils.py but I could not find something that makes much more sense than putting it here. I agree anyway that it should not be public.